src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -138,15 +145,28 @@ public static bool IsPathRooted(string path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
         {
-            if (path == null) return null;
+            if (path == null)
+                return null;
+
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
-            // Need to return the normalized directory separator
             path = PathInternal.NormalizeDirectorySeparators(path);
+            ReadOnlySpan<char> result = GetPathRoot(path.AsReadOnlySpan());
+
+            if (path.Length == result.Length && object.ReferenceEquals(result.DangerousGetPinnableReference(), path))
```

Comment:
Sorry, I spaced a bit when I wrote this earlier. You need a helper like this:

``` C#
unsafe static bool AreSameString(string value, ReadOnlySpan<char> span)
{
    if (value.Length != span.Length)
        return false;

    fixed (char* vp = value)
    {
        fixed (char* sp = &span.DangerousGetPinnableReference())
            return vp == sp;
    }
}
```

I don't think we have any other better ways to do this, but I'm not sure. @stephentoub, are you aware of existing helpers or a better way to check a char span's target for reference equality to an existing string?

Reply:
how is it different from  ```  object.ReferenceEquals(result.DangerousGetPinnableReference(), path))``` ?

Reply:
> object.ReferenceEquals(result.DangerousGetPinnableReference(), path))

This creates a boxed copy of the first character in result; and compares the boxed copy to path. It will always return  false. You may want to create a test that verifies that there was actually no allocation when one was not expected.

Reply:
(Overlaps Span extension method would work too, but it is not available in CoreLib today.)

Reply:
@jkotas  I am not able to understand the suggestion properly. Unsafe.safe is not available in coreclr.
```
 internal static bool AreSame(string value, ReadOnlySpan<char> span)
 {
        if (value.Length != span.Length)
             return false;

         return object.ReferenceEquals(value, MemoryMarshal.GetReference(span));
 }
```
are you saying something like this ?

Reply:
Unsafe is available in CoreLib, but I do not think you want to use it here as I have said.

Reply:
okay. is the code I suggested above correct ?

Reply:
No, it is not. It will always return false.